perf/bloom_filter/bloom_filter.cc

```diff
@@ -0,0 +1,297 @@
+		for (std::uint32_t i = 0; i < tuple_count; ++i) {
+			Tuple newTuple(i);
+			insertTuples.push_back(newTuple);
+			selectTuples.push_back(newTuple);
+		}
+		std::cout << "Finished filling tupleData vectors." << std::endl;
+		std::cout << std::endl;
+	}
+
+	TupleData(const TupleData &) = delete;
+
+	TupleData(TupleData &&) = delete;
+
+	~TupleData() {
+		for (auto & t : selectTuples)
+			free(t.begin);
+	}
+
+	std::vector<Tuple> selectTuples;
+	std::vector<Tuple> insertTuples;
+};
+
+/** Tuple that will make select query look up all tree. */
+Tuple *nonexistenetTuple;
+/** Struct that holds:
+ *  1. inserted tuples;
+ *  2. tuples for select queries.
+ */
+TupleData *data;
+
+/** Args passed through lua code. */
+struct BenchArgs {
+	/** Name of benchmarked space. */
+	const char *space_name;
+	/** Space id. */
+	uint32_t space_id;
+	/** Tuple count. */
+	uint32_t tuple_count;
+
+	explicit BenchArgs(const char *args, const char *args_end) {
+		uint32_t arg_count = mp_decode_array(&args);
+		uint32_t name_len = 0;
+		const char *name = mp_decode_str(&args, &name_len);
+		space_name = strndup(name, name_len);
+		space_id = mp_decode_uint(&args);
+		tuple_count = mp_decode_uint(&args);
+	}
+
+	BenchArgs(const BenchArgs &) = delete;
+
+	BenchArgs(BenchArgs &&other) = delete;
+
+	~BenchArgs() { free((void *)space_name); }
+};
+
+/** Trivial wrapper for storing benchmark execution time. */
+struct BenchTimeResult {
+	double elapsed_time;
+};
+
+/** Represents results of benchmark. */
+struct BenchAlgorithmResult {
+	BenchAlgorithmResult(const char *name) : space_name(name) {}
+
+	/** Name of benchmarking space. */
+	const char *space_name;
+	/** Primary index bloom filter binary size. */
+	size_t bloom_bsize = 0;
+	struct BenchTimeResult insertTimeMs;
+	struct BenchTimeResult selectTimeMs;
+};
+
+int
+init(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	uint32_t arg_count = mp_decode_array(&args);
+	uint32_t name_len = 0;
+	const char *name = mp_decode_str(&args, &name_len);
+	const char *build_name = strndup(name, name_len);
+	uint32_t tuple_count = mp_decode_uint(&args);
+
+	(void)ctx;
+	(void)args;
+	(void)args_end;
+
+	std::cout << std::endl;
+	std::cout << "-----------------------------------------------------------------" << std::endl;
+	std::cout << "STARTED benchmarking." << std::endl;
+
+	memset(tuple_string, '.', STRING_FIELD_SIZE);
+	nonexistenetTuple = new Tuple(tuple_count + 1);
+	data = new TupleData(tuple_count);
+	resultsFile.open(resultsFileName, std::ios_base::in);
+	if (!resultsFile) {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does not exist. Creating it." << std::endl;
+		resultsFile.open(resultsFileName, std::ios_base::app);
+		resultsFile << "builds:" << std::endl;
+	} else {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does exist. Starting to write." << std::endl;
+		resultsFile.close();
+		resultsFile.open(resultsFileName, std::ios_base::app);
+	}
```

Comment:
This code looks redundant, why not use `std::ios_base::app` from the beginning?

Reply:
If I understand it right, `std::ios_base::app`would always think that file exists, but we need to add _builds:_ line in case it doesn't.